core/liquidity/v2/sla.go

```diff
@@ -35,14 +49,14 @@ func (e *Engine) BeginBlock(txs []TX) {
 	e.kSla = e.GenerateKSla(txs)
 }
 
-func (e *Engine) TxProcessed(txCount int) {
+func (e *Engine) TxProcessed(txCount int, markPrice *num.Uint, positionFactor num.Decimal) {
 	// Check if the k transaction has been processed
-	if e.kSla != txCount {
+	if e.kSla != txCount && e.auctionState.IsOpeningAuction() {
```

Comment:
let's remove the randomness/k/ txProcessed and just do it at the end of block as agreed.